plugin/documents.py

```diff
@@ -587,6 +611,14 @@ def _resolve_visible_code_lenses_async(self) -> None:
                 callback = functools.partial(self._on_resolved_code_lens_async, session.config.name, index, region)
                 session.send_request_async(Request("codeLens/resolve", code_lens, self.view), callback)
 
+        # To prevent excessive jumping around, it would be desirable to only call
+        # phantom_set.update when most of the data is already resolved.
+        # Since this function is called whenever code lenses are actually rendered
+        # this function is a good place to put the actual rendering after some delay
+        # to wait for the code lenses to resolve.
+        if userprefs().show_code_lens == "phantom":
+            sublime.set_timeout_async(self._show_code_lens_phantom, FEATURES_TIMEOUT)
+
```

Comment:
How about using `Promise.all` in plugin/core/promise.py to wait for all lenses in the viewport to resolve?

Reply:
Would also prefer to have the same resolvement logic for both annotations and phantoms

Reply:
So basically use `session.send_request_task` and collect them all in a list and use `Promise.all`. See textDocument/completion

Reply:
That's a good suggestion. I'll incorporate it in my next set of patches.